### PR TITLE
extracting function signatures

### LINKED ISSUE #457: Developers SBAT fix `extractFunctionSignature` function
### Expected Behavior

`extractFunctionSignature` returns the correct value, we have a test that verifies this

### Current Behavior

`extractFunctionSignature` does not return the correct value

### PR DESCRIPTION
**Pull request by mrsmkl**
### Description

Tested the function, it didn't seem to work, so copied from https://ethereum.stackexchange.com/questions/61826/how-to-extract-function-signature-from-msg-data instead.

### Tested

Just added one test case for the function.

### Other changes

Moved the function into library.

### Related issues

- Fixes #457

### CODE REVIEW

File: packages/protocol/test/common/extractfunctionsignature.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import {
+  ExtractFunctionSignatureTestContract,
+  ExtractFunctionSignatureTestInstance,
+  TestTransactionsContract,
+  TestTransactionsInstance,
+} from 'types'
+// import { stripHexEncoding } from '../../lib/test-utils'
```
Comment (m-chrzan) [inline packages/protocol/test/common/extractfunctionsignature.ts]: Remove commented out code.

File: packages/protocol/test/governance/governance.ts [hunk 2]
```diff
@@ -82,9 +82,18 @@ contract('Governance', (accounts: string[]) => {
   const baselineQuorumFactor = 1
   const weight = 100
   const participation = 1
-  const expectedParticipationBaseline = fromFixed(baselineUpdateFactor)
+  /*  const expectedParticipationBaseline = fromFixed(baselineUpdateFactor)
```
Comment (m-chrzan) [inline packages/protocol/test/governance/governance.ts]: Why did this file have to change?

File: packages/protocol/contracts/common/ExtractFunctionSignature.sol [hunk 3]
```diff
@@ -0,0 +1,13 @@
+pragma solidity ^0.5.3;
+
+library ExtractFunctionSignature {
```
Comment (m-chrzan) [inline packages/protocol/contracts/common/ExtractFunctionSignature.sol:3]: I'm wondering about the usefulness of this being a library. Do we use this in any other contracts?

If we do keep it a library, let's make the function `internal`, not `public`, and not link it during migrations.

Comment (mrsmkl) [reply packages/protocol/contracts/common/ExtractFunctionSignature.sol:3]: Mostly it's in a library to make testing it easier...

File: packages/protocol/migrationsConfig.js [hunk 4]
```diff
@@ -99,6 +99,7 @@ const linkedLibraries = {
   IntegerSortedLinkedList: ['Governance', 'IntegerSortedLinkedListTest'],
   AddressSortedLinkedListWithMedian: ['SortedOracles', 'AddressSortedLinkedListWithMedianTest'],
   Signatures: ['LockedGold', 'Escrow'],
+  //  ExtractFunctionSignature: ['Governance', 'ExtractFunctionSignatureTest'],
```
Comment (asaj) [inline packages/protocol/migrationsConfig.js]: Remove?

**Review by asaj — APPROVED** [review]
Nice!

**Pull request merged**